include/aca_dhcp_server.h

```diff
@@ -60,8 +71,74 @@ struct dhcp_entry_data {
 #define DHCP_MSG_HWTYPE_ETH (0x1)
 #define DHCP_MSG_HWTYPE_ETH_LEN (0x6)
 #define DHCP_MSG_MAGIC_COOKIE (0x63825363) //magic cookie per RFC2131
+#define DHCP_MSG_SERVER_ID (0x7f000001) //Hard coded for 127.0.0.1
 
 #pragma pack(push, 1)
+
+// define ip header
+struct iphear {
+  uint8_t version;
+  uint8_t ds;
+  uint16_t total_len;
+  uint16_t identi;
+  uint16_t fregment;
+  uint8_t tol;
+  uint8_t protocol;
+  uint16_t checksum;
+  uint32_t src_ip;
+  uint32_t dst_ip;
+  uint16_t src_port;
+  uint16_t dst_port;
+  uint16_t len;
+  uint16_t udp_checksum;
+};
+
+// define upd header
+struct udphear {
+  uint32_t srcIp;
+  uint32_t dstIp;
+  uint16_t udp_len;
+  uint8_t rsv;
+  uint8_t protocol;
+  uint16_t src_port;
+  uint16_t dst_port;
+  uint16_t len;
+  uint16_t checksum;
+  uint8_t op;
+  uint8_t htype;
+  uint8_t hlen;
+  uint8_t hops;
+  uint32_t xid;
+  uint16_t secs;
+  uint16_t flags;
+  uint32_t ciaddr;
+  uint32_t yiaddr;
+  uint32_t siaddr;
+  uint32_t giaddr;
+  uint8_t chaddr[16];
+  uint8_t sname[64];
+  uint8_t file[128];
+  uint32_t cookie;
+  uint8_t options[308];
+};
+
+//DHCP message ip header field
+#define DHCP_MSG_IP_HEADER_SRC_IP (0x7f000101) // hard code for dhcp src ip 127.0.1.1
+#define DHCP_MSG_IP_HEADER_DEST_IP (0xffffffff)
+#define DHCP_MSG_IP_HEADER_SRC_PORT (67)
+#define DHCP_MSG_IP_HEADER_DEST_PORT (68)
+#define DHCP_MSG_IP_HEADER_VERSION (69) //ip version 4 + ip header length 0100 + 0101(length is base 4 bytes, all length=5*4=20)
```

Comment:
okay... so how did it become "69"?

Reply:
We will be ready to merge after addressing this last comment.

Reply:
The value 69 corresponds to 0x45 in hexadecimal notation and 01000101. The value is 0100 (ip version 4), 0101(ip header length 5), and then 5 times 4 is the IP header length 20.

Reply:
got it. Thanks!